src/app/content/utils/urlUtils.ts

```diff
@@ -28,11 +30,16 @@ export const getBookPageUrlAndParams = (
   return {params, state, url: contentRoute.getUrl(params) + query};
 };
 
-export const getUrlParamsForBook = (
-  book: Pick<Book, 'id' | 'tree' | 'title' | 'version'> & Partial<{slug: string}>
-): Params['book'] => {
-  if ('slug' in book && book.slug && BOOKS[book.id]) {
-    return book.version === BOOKS[book.id].defaultVersion
+export const getUrlParamsForBook = async(
+  book: Pick<Book, 'id' | 'tree' | 'title' | 'version'> & Partial<{slug: string}>,
+  bookConfigLoader?: AppServices['bookConfigLoader']
+): Promise<Params['book']> => {
```

Comment:
I think there should be a way to declare this function and `getBookPageUrlAndParams` to be synchronous if `bookConfigLoader` is not passed and `async` if it is
This way we could use the same function for both cases

Reply:
@PiotrKozlowski 
I was thinking about that, but i'm not sure how to make the function to be optional async. Maybe by returning the result in `Promise.resolve` if `bookConfigLoader` is passed? I was experimenting with this approach and it makes the function unintelligible, but maybe it can be improved.

Reply:
You can try something like this:
```ts
function a(b: 1): 1;
function a(b: 1, v: boolean): Promise<1>;
function a(b: 1, v?: boolean) {
  if (v) {
    return Promise.resolve(b)
  }
  return b
}

const z = a(1);
const zz = a(1, true);
```

Reply:
it is called Function Overloading - theoretically number of arguments should be always the same, but you can experiment with it

Reply:
Yeah, i know what function overloading is. I thought that there is no function overloading in js, but as I understand _here comes the typescript!_